### DISCUSSION TITLE
How to deploy winui app without Code Signing certificate?

### DISCUSSION THREAD

**Discussion by steam3d** [microsoft/microsoft-ui-xaml#9152] [Q&A] [answered]
Hi. For individual developers, it is almost impossible to obtain a trusted certificate, and if it is possible, it costs almost $400 per year.
Is there a way to deploy an application to users without signing the code? Like in WPF I just copied *.exe

**Comment by DarranRowe** [comment, marked as answer]
The certificate is really only there for the .msix package. So the easiest option is to not use the package. [Create your first WinUI 3 project](https://learn.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app) gives the steps for making a project unpackaged.

If you want both the .msix package and no code signing certificate, then you are out of luck. Windows needs the package to be signed by a trusted certificate before you install it. If you are in a limited business environment or you are distributing these applications to a very small group of people who trust you, then you can use either a business certificate from the business' CA, or a self signed certificate.

If you do set the project to unpackaged, you should be able to copy the contents of the output directory for C#, I know you can with C++.

**Reply by steam3d** [reply]
So for Microsoft Store I can build packaged version of an app and for outside Microsoft Store deliver I can build unpackaged app (that will not require certificate and user could easily install it), right?

**Reply by DarranRowe** [reply]
Yes, to be clear on one point, the application would be dependent on some redistributables at this point. C++ and C# applications would both depend upon the Windows App SDK redistributable. C# applications would depend on the .NET runtime. C++ applications would depend on the VC redistributable. This is unavoidable since these components are not distributed as part of Windows.
It is possible to make the application self contained, but you are just exchanging the redistributable dependencies for your application's apparent install size.

**Reply by llothar** [reply]
Why can't we have something like LetsEncrypt for software. The certification process is just plain road robbery and offers no benefits at all. It's enough to save domain signed entities. No further validation should be required, at least not for the renew certifications. Please Windows, do some notarization like Apple does it with a reasonable fee.
